### DISCUSSION TITLE
RFID Scan not working though the webapp.

### DISCUSSION THREAD

**Discussion by SubLim3JS** [MiczFlor/RPi-Jukebox-RFID#1998] [Q&A] [answered]
I've got a Pi3 imaged from the official Pi Imager (Bullseye 32bit). Attached are the steps I took to do the setup. Everything seems to be working just fine but the webapp doesn't acknowledge the RFID scan. The web page loads, recognizes audio files and folders and plays music without an issue. I know the RC522 works because I can run basic scripts (attached) and it works fine.

Any help is appreciated. Let me know if you need any kind of additional information. Thanks.

[Setup Steps.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11046529/Setup.Steps.txt)
[RC522 Test Scripts.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11046530/RC522.Test.Scripts.txt)

**Comment by AlvinSchiller** [comment]
Hi,
can you maybe provider your Installation Log (/home/pi/phoniebox_logs/) and the systemlog (tail /var/log/syslog -n 200)?
And the Status of the rfid reader service (systemctl status phoniebox-rfid-reader.service)?

Also this sound pretty similiar to https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1972

**Comment by SubLim3JS** [comment]
Sure thing. Attached.

[RFID Reader Services.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11056980/RFID.Reader.Services.txt)
[phoniebox_logs.log](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11056981/phoniebox_logs.log)

**Reply by s-martin** [reply]
Seems like cloning into the repo at least caused a warning

```
Cloning into 'RPi-Jukebox-RFID'...
Installing additional Python packages...
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))':
```

and therefore installing of python packages did not work correctly.
Maybe the server was not available at that time.

The are a couple of errors following when setting the user in samba and you selected the reader.

```
Continue (y/n)? Installing Python requirements for RC522...
ERROR: Could not open requirements file: [Errno 2] No such file or directory: '/home/pi/RPi-Jukebox-RFID/components/rfid-reader/RC522/requirements.txt'
Activating SPI...
Configure RFID reader in Phoniebox...
cp: cannot stat '/home/pi/RPi-Jukebox-RFID/scripts/Reader.py.experimental': No such file or directory
/home/MagiCube/RPi-Jukebox-RFID/components/rfid-reader/RC522/setup_rc522.sh: line 27: /home/pi/RPi-Jukebox-RFID/scripts/deviceName.txt: No such file or directory
chown: invalid user: âpi:www-dataâ
chmod: cannot access '/home/pi/RPi-Jukebox-RFID/scripts/deviceName.txt': No such file or directory
```

Did you retry the installation script?

**Comment by SubLim3JS** [comment]
I've tried before with no luck. I tried again just now using this same image. I'm going to try this once more from a clean image (Bullseye with update/upgrade). Is there anything from my install steps attached earlier that would be an issue? Any reason I shouldn't change the hostname, set Wifi/local etc from the Official Pi Imager tool?

**Comment by SubLim3JS** [comment]
No Luck. One of the errors says "Failed to add entry for user pi." so I wonder if it's the hostname. I'm going to try and fix that and test again.

[RFID Reader Services 2.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057381/RFID.Reader.Services.2.txt)
[phoniebox_logs 2.log](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057382/phoniebox_logs.2.log)

**Comment by SubLim3JS** [comment]
Still no love. Brand new Image from Pi Imager. Followed my Install Setup doc. Update/Upgrade. Then ran the one-line script. That did not read the RFID. (See logs 3)

Then ran the script again. See logs 4. Let me know what else I should try. Thanks.

[Install Setup.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057594/Install.Setup.txt)

[phoniebox_logs_3.log](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057595/phoniebox_logs_3.log)
[RFID Reader Services 3.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057596/RFID.Reader.Services.3.txt)

[phoniebox_logs_4.log](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057640/phoniebox_logs_4.log)
[RFID Reader Services 4.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11057641/RFID.Reader.Services.4.txt)

**Comment by AlvinSchiller** [comment]
-> "/home/**MagiCube**/RPi-Jukebox-RFID/"
Phoniebox needs the user "pi" to work. This is an open issue, but theres no workaround at the moment (https://github.com/MiczFlor/RPi-Jukebox-RFID/issues/1785). 
But you it seems you have changed that already in the last installations.

Have you tried skipping all the manual installations in advance (even the spi / i2c activations)? The Installationscript / plugin setup should make all needed configurations.
Also have you tried using the bullseye **light** image or even buster?

**Reply by AlvinSchiller** [reply]
I tried the installations on bullseye **lite** and **full**, with and without your preinstall steps (all with a fresh image), and in all cases the Id is updated in the WebUi at "Last used Chip ID".

Can you again post the service status directly after you swiped a card?

**Reply by AlvinSchiller** [reply]
Also can you make sure that the reader is connected correctly with all 8 Pins?
https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Wiring_for_RC522_card_reader

**Reply by SubLim3JS** [reply]
I tested on a new image last night (same setup steps) but kept the user as "pi". Still not registering. I have not tried this on a light OS but did try Buster before hopping in here for extra support. I'm going to start a new image and skip my manual setup steps. I'll update/upgrade and run the one-liner.

Question, if I'm still unsuccessful would you be able to send me a known working image I can flash? I'm using an 8GB SD. If a known working image doesn't work then I'm definitely doing something wrong. Since the basic test scripts work to read and write the RFID I can't see that being the case. It is wired correctly per that link and the attached pinout.

Here is the latest RFID reader services. 
[RFID Reader Services 5.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11063907/RFID.Reader.Services.5.txt)
![RFID-RC522-Pinout](https://user-images.githubusercontent.com/24327186/227571238-d4346ad0-fc23-4bae-aa9c-2f74d0200d1e.png)

**Reply by AlvinSchiller** [reply]
What do you mean with "IRQ <-> unused"?
IRQ needs to be connnected to Pin 18/GPIO 24, otherwise it won't work

**Comment by SubLim3JS** [comment]
Still no love. It has to be something stupid I just can't pin down.

[RFID Reader Services 6.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11066917/RFID.Reader.Services.6.txt)
[phoniebox_logs 6_2.log](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11066918/phoniebox_logs.6_2.log)
[phoniebox_logs 6_1.log](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11066919/phoniebox_logs.6_1.log)
[systemlog 6.txt](https://github.com/MiczFlor/RPi-Jukebox-RFID/files/11066920/systemlog.6.txt)

**Comment by SubLim3JS** [comment, marked as answer]
OMFG!!! It was the IRQ that was needed! Every Google search and RC522 pinout document shows that pinout, with the IRQ not needed.... I added that 1 pin and sure enough I got the response needed.

I'm truly sorry I didn't catch that sooner! Gahhh. Time to start with a fresh image and go from there. Thanks for all the help!

**Reply by AlvinSchiller** [reply]
Glad it works now 👍

**Reply by ConsolasTrovao** [reply]
Hey my friend! I'm having the same issue and I do have the IRQ disconnected as well so fingers cross it is the same thing - where should I connect it to (the pin in the PI)? 

Thanks!

**Reply by AlvinSchiller** [reply]
Hi, have a look at this thread:
https://github.com/MiczFlor/RPi-Jukebox-RFID/discussions/1998#discussioncomment-5415811

There is a link to the wiki documentation, and further description.

**Reply by ConsolasTrovao** [reply]
Outstanding! Will check it! Thank you so much, Alvin!

**Reply by ConsolasTrovao** [reply]
And magic, it works (of course!). Thank you!

**Comment by SubLim3JS** [comment]
Hey. Here is what I ended up with. IRQ on pin 18 worked.

[cid:3b24cdbb-d831-4e0d-be3d-68a61c97b855]
________________________________
